### PR TITLE
feat(common): Add 'syncTimeout' and 'forceSync' to LoadOpts

### PR DESCRIPTION
**Pull request by stbrody**

### CODE REVIEW

File: packages/core/src/state-management/state-manager.ts [hunk 1]
```diff
@@ -95,14 +97,17 @@ export class StateManager {
     if (publish) {
       this.publishTip(state$);
     }
-    const tip$ = this.dispatcher.messageBus.queryNetwork(state$.id);
     if (sync) {
-      const tip = await tip$.pipe(timeoutWith(3000, of(undefined))).toPromise();
+      const tip$ = this.dispatcher.messageBus.queryNetwork(state$.id);
+
+      const syncTimeout = (opts as LoadOpts).syncTimeoutMillis != undefined ? (opts as LoadOpts).syncTimeoutMillis : DEFAULT_SYNC_TIMEOUT
+      if (syncTimeout == 0) { // timeout of zero means don't wait for the response at all
```
Comment (stbrody) [inline packages/core/src/state-management/state-manager.ts:104]: The rxjs docs don't seem to specify how `timeoutWith` handles a timeout of `0`, so I added this just to be safe, and to make the behavior explicit

Comment (ukstv) [reply packages/core/src/state-management/state-manager.ts:104]: It handles it fine.

File: packages/core/src/ceramic.ts [hunk 2]
```diff
@@ -67,9 +67,9 @@ const SUPPORTED_CHAINS_BY_NETWORK = {
   [Networks.INMEMORY]: ["inmemory:12345"], // Our fake in-memory anchor service chainId
 }
 
-const DEFAULT_APPLY_COMMIT_OPTS = { anchor: true, publish: true, sync: false }
-const DEFAULT_CREATE_FROM_GENESIS_OPTS = { anchor: true, publish: true, sync: true }
-const DEFAULT_LOAD_OPTS = { sync: true }
+const DEFAULT_APPLY_COMMIT_OPTS = { anchor: true, publish: true, sync: false, forceSync: false }
```
Comment (oed) [inline packages/core/src/ceramic.ts:70]: Can we name it to something more descriptive like `forceSyncIfCached`?

File: packages/core/testOutput.txt [hunk 3]
```diff
@@ -0,0 +1,134 @@
+(node:182562) V8: /home/spencer/js-ceramic4/node_modules/ipld-dag-cbor/node_modules/borc/src/decoder.asm.js:3 Linking failure in asm.js: Unexpected stdlib member
+(Use `node --trace-warnings ...` to show where the warning was created)
```
Comment (oed) [inline packages/core/testOutput.txt:2]: Should this file really be here?

**Comment by ukstv** [comment]
Apparently, my original suggestion "sync is enough" incredibly slows down the computation.

**Comment by ukstv** [comment]
But maybe extend current `sync` from mere boolean flag to string-based variable like https://github.com/ceramicnetwork/js-ceramic/tree/su/sync-opts-variability ?

File: packages/core/testOutput.txt (hunk 3 above)
Comment (stbrody) [reply packages/core/testOutput.txt:2]: Whoops!  Sorry about that

**Comment by stbrody** [comment]
Closed in favor of https://github.com/ceramicnetwork/js-ceramic/pull/1269

**Pull request closed without merging**